cider/exec/template/DateTimeIR.cpp

```diff
@@ -74,8 +74,36 @@ const char* get_extract_function_name(ExtractField field) {
   return "";
 }
 
+bool is_extract_function_complex(ExtractField field) {
+  switch (field) {
+    case kEPOCH:
+    case kDATEEPOCH:
+    case kQUARTERDAY:
+    case kHOUR:
+    case kMINUTE:
+    case kSECOND:
+    case kMILLISECOND:
+    case kMICROSECOND:
+    case kNANOSECOND:
+      return false;
+    case kDOW:
+    case kISODOW:
+    case kDAY:
+    case kWEEK:
+    case kWEEK_SUNDAY:
+    case kWEEK_SATURDAY:
+    case kDOY:
+    case kMONTH:
+    case kQUARTER:
+    case kYEAR:
+      return true;
+  }
+  UNREACHABLE();
```

Comment:
throw cider exception instead?

Reply:
During substrait to analyzer translation process, this part is already checked in  https://github.com/intel/BDTK/blob/main/cider/function/datetime/ExtractFromTime.cpp#L301

Reply:
double thought, add ```cider throw``` here may help future development. updated